src/components/Library.vue

```diff
@@ -175,52 +180,85 @@ export default {
     * CREATE OBJECT
     * initializes an empty object
     */
-    createObject() {
-      switch (this.mode) {
-        case 'stories':
-          this.$store.dispatch('models/initStory');
-          return;
-        case 'spaces':
-          this.$store.dispatch('models/initSpace', { story: this.currentStory });
-          break;
-        case 'shading':
-          this.$store.dispatch('models/initShading', { story: this.currentStory });
-          break;
-        case 'images':
-          window.eventBus.$emit('uploadImage');
-          break;
-        case 'windows':
-        case 'daylighting_controls':
-          window.eventBus.$emit('error', 'Create components by clicking where you would like it to be');
-          break;
-        default:
-          this.$store.dispatch('models/createObjectWithType', { type: this.mode });
-          break;
+    createObject({ duplicate }) {
```

Comment:
I'm getting some errors here - I think this needs a default for when the argument isn't passed in.

Reply:
Sounds good. I wasn't hitting any errors in my testing. What steps were you using to get the errors?

Reply:
Added a default value here. Let me know if you're still erroring out.

Reply:
For my test I

1. Created a new floorplan
2. Created a square using the default space on that plan
3. Clicked copy on the left-hand panel to copy it

Reply:
Thanks. Changed it to be a simple param instead of passing it as an object. Seems to be fixed.

Reply:
This should no longer error out.